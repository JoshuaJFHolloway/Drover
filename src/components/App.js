import React, {Component} from "react";
import TextInputBox from "./TextInputBox";
import ScaleInputBox from './ScaleInputBox';
import Results from './Results';
import Logo from './Logo';

const API_URL = 'https://app.joindrover.com/api/web/vehicles';

const columnStyle = { display: 'inline-block', width: '50%', 'vertical-align': 'top' };

class App extends Component {
  constructor() {
    super();

    this.state = {
      params: {
        max_distance: 50,
        number_of_months: 12,
        number_of_weeks: 52,
        order_by: "price",
        order_direction: "asc",
        page: 1,
        per_page: 15,
        price_max: 2500,
        price_min :100,
        rolling: false,
        start_date: new Date(Date.now()).toLocaleString().slice(0, -10),
        vehicle_make: "Any",
        transmission: "Any",
        year: "Any", // integer
        fuel: "Any",
        tags: "Any", // array

        vehicle_type: 'Consumer',
        location: 'London, UK',
      },
      results: [],
      noResults: false
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSlider = this.handleSlider.bind(this);
  }

  componentWillMount() {
    this.getData();
  };

  areParamsValid() {
    const state = this.state.params;

    if((state.start_date.length &&
        state.location.length &&
        state.max_distance.toString().length &&
        state.transmission.length &&
        state.vehicle_make.length &&
        state.fuel.length &&
        state.tags.length &&
        state.year.toString().length) > 0){
      return true;
    }
  }

  distill(data){
    const body = {};
    const keys = Object.keys(data);

    keys.forEach(key => {
      const value = data[key];
      if(value !== "Any") {
        body[key] = value;
      }
    });

    return body;
  }

  getData() {
    if (this.areParamsValid()) {
      const body = this.distill(this.state.params);
      console.log("Fetching Data");

      const apiConfig = {
        method: 'post',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
        },
        body: JSON.stringify(body)
      };

      fetch(API_URL, apiConfig)
        .then(results => results.json())
        .then(response => {
          console.log(response);
          this.setState({
            results: response.data
          });
          if (response.data.length === 0) {
            this.results(true)
          }
          else (this.results(false))
        })
    }
  }

  results(areResultsPresent) {
    this.setState({
      noResults: areResultsPresent,
    });
  }

  handleChange(event) {
    const { name, value } = event.target;
    const params = this.state.params;

    if(name === "max_distance") {
      params[name]= parseInt(value);
      this.setState({params}, this.getData());
    }
    if(name === "year") {
      params[name]= parseInt(value);
      this.setState({params}, this.getData());
    }
    // if(event.target.name === "tags"){
    //   params[name]= [event.target.value];
    //   this.setState({params}, this.getData());
    // }
    else {
      params[name] = value;
      this.setState({params}, this.getData());
    }
  }

  handleSlider(event) {
    const params = this.state.params;
    params["price_min"] = event[0];
    params["price_max"] = event[1];

    this.setState({params}, this.getData());
  }

  handleSelectChange(event) {
    debugger;
    console.log(event);
  }

  render() {
    const {params} = this.state;
    if (this.state.noResults === false && this.state.results.length < 1) {
      return <span>Loading Cars...</span>
    }

    const resultsArray = [];
    if(this.state.noResults === false){
      const cap = this.state.results.length < 14 ? this.state.results.length : 14;

      for (let i = 0; i < cap; i++) {
        const image = this.state.results[i].images.find(i => i.position === 0);
        const otherImages = this.state.results[i].images.sort((a, b) => a.position - b.position).slice(1, 4);

        resultsArray.push(
          <Results
            vehicleMake={this.state.results[i].vehicle_make}
            vehicleModel={this.state.results[i].vehicle_model}
            postcode={this.state.results[i].postcode}
            mainImage={image.main_image_url}

            features={this.state.results[i].features} // only take a maximum of 4 features
            fuel={this.state.results[i].fuel}
            seats={this.state.results[i].number_seats_information}
            engineSize={this.state.results[i].engine_size_information}
            price={this.state.results[i].price_discount_and_deposit_schedule_hash[1].driver_price_pounds_after_discount_including_insurance}
          />
        )
      }
    }

  return (
    <div>
      <div style={columnStyle}>
        <Logo
          title={"Drover"}/>
        <form>
        <TextInputBox
          eventHandler={this.handleChange}
          handleSelectChange={this.handleSelectChange}
          name={"start_date"}
          title={"Subscription Start"}
          value={params.start_date}
          type={"text"}
        />
        <TextInputBox
          eventHandler={this.handleChange}
          name={"location"}
          placeholder={"Enter your location"}
          title={"Location"}
          value={params.location}
          type={"text"}
        />
        <TextInputBox
          eventHandler={this.handleChange}
          name={"max_distance"}
          title={"Distance (radius in miles)"}
          value={params.max_distance}
          type={"number"}
        />
        <ScaleInputBox
          eventHandler={this.handleSlider}
          title={"Monthly Budget"}
          value= {params.location}
          min= {this.state.params.price_min}
          max= {this.state.params.price_max}
        />
        <TextInputBox
          eventHandler={this.handleChange}
          name={"vehicle_make"}
          title={"Vehicle Make"}
          value={params.vehicle_make}
          type={"text"}
        />
        <TextInputBox
          eventHandler={this.handleChange}
          name={"transmission"}
          title={"Gear Box"}
          value={params.transmission}
          type={"text"}
        />
        <TextInputBox
          eventHandler={this.handleChange}
          name={"year"}
          title={"Year"}
          value={params.year}
          type={"number"}
        />
        <TextInputBox
          eventHandler={this.handleChange}
          name={"fuel"}
          title={"Fuel Type"}
          value={params.fuel}
          type={"text"}
        />
        <TextInputBox
          eventHandler={this.handleChange}
          name={"tags"}
          title={"Car Type"}
          value={params.tags}
          type={"text"}
        />
        </form>
      </div>
      <div style={columnStyle}>
        {resultsArray}
      </div>
    </div>
    )
  }
}
export default App;