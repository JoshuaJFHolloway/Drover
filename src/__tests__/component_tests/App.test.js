import React from 'react';
import { shallow } from 'enzyme';
import App from '../../components/App';

const params = {
  subscription_start_days: 30,
    location: 'London, UK',
    max_distance: 50,
    vehicle_make: "Any",
    price_min :100,
    transmission: "Any",
    year: new Date().getFullYear(), // integer
    fuel: "Any",
    tags: "Any", // array
    price_max: 2500,

    number_of_months: 12,
    number_of_weeks: 52,
    order_by: "price",
    order_direction: "asc",
    page: 1,
    per_page: 15,
    rolling: false,
    vehicle_type: 'Consumer',
};

const handleChangeEvent = {
  target: {
    name: 'fuel',
    value: 'petrol'
  }
};

const handleSliderEvent = [300, 2000];

describe('App', () => {
  let app = shallow(<App/>);

  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  describe('passing props', () => {

    describe('InputsHolder component', () => {

      it('has state', () => {
        expect(app.find('InputsHolder').prop('state')).toBe(app.instance().state);
      });

      it('has scaleHandler', () => {
        expect(app.find('InputsHolder').prop('scaleHandler')).toBe(app.instance().handleSlider);
      });

      it('has textHandler', () => {
        expect(app.find('InputsHolder').prop('textHandler')).toBe(app.instance().handleChange);
      });

      it('has capitalize', () => {
        expect(app.find('InputsHolder').prop('capitalize')).toBe(app.instance().capitalize);
      });
    });

    describe('Logo component', () => {

      it('has title', () => {
        expect(app.find('Logo').prop('title')).toBe(app.instance().title);
      })
    });

    describe('ResultsHolder component', () => {

      it('has state', () => {
        expect(app.find('ResultsHolder').prop('state')).toBe(app.instance().state);
      });

      it('has capitalize', () => {
        expect(app.find('ResultsHolder').prop('capitalize')).toBe(app.instance().capitalize);
      })
    });

  });

  describe('Initializes state', () => {

    it('initializes state correctly', () => {
      expect(app.state('params')).toEqual(params);
    });
  });

  describe('Updating state successfully', () => {

    describe('handleChange', () => {

      it('updates the state correctly depending on the event.target', () => {
        app.instance().handleChange(handleChangeEvent);
        expect(app.state('fuel')).toEqual('petrol');
      });
    });

    describe('handleSlider', () => {

      it('updates the price correctly', () => {
        app.instance().handleSlider(handleSliderEvent);
        expect(app.state('price_min')).toEqual(300);
        expect(app.state('price_max')).toEqual(2000);
      });
    })
  });

  describe('getData', () => {

    it('calls the drover url and returns data to me', () => {
      return app.getData()
        .then(data => {
          expect(data).toBeDefined();
          expect(data.owner_name).toEqual('Drover Ltd')
        });
    })
  })

});

//   describe('getData', () => {
//
//     beforeEach(() => {
//       fetch.resetMocks()
//     });
//
//     it('calls the drover url and returns data to me', () => {
//       fetch.mockResponseOnce(JSON.stringify({results: 'cars'}));
//
//       APIRequest('google').then(res => {
//         expect(res.results).toEqual('cars')
//       });
//
//       expect(fetch.mock.calls.length).toEqual(1);
//       expect(fetch.mock.calls[0][0]).toEqual('https://app.joindrover.com/api/web/vehicles');
//     });
//   })
//
// });

