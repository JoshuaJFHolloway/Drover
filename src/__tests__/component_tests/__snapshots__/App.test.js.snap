// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          style={
                    Object {
                              "display": "inline-block",
                              "verticalAlign": "top",
                              "width": "50%",
                            }
          }
>
          <Logo
                    title="Drover"
          />
          <form>
                    <InputsHolder
                              capitalize={[Function]}
                              scaleHandler={[Function]}
                              state={
                                        Object {
                                                  "noResults": false,
                                                  "params": Object {
                                                    "fuel": "Any",
                                                    "location": "London, UK",
                                                    "max_distance": 50,
                                                    "number_of_months": 12,
                                                    "number_of_weeks": 52,
                                                    "order_by": "price",
                                                    "order_direction": "asc",
                                                    "page": 1,
                                                    "per_page": 15,
                                                    "price_max": 2500,
                                                    "price_min": 100,
                                                    "rolling": false,
                                                    "subscription_start_days": 30,
                                                    "tags": "Any",
                                                    "transmission": "Any",
                                                    "vehicle_make": "Any",
                                                    "vehicle_type": "Consumer",
                                                    "year": 2018,
                                                  },
                                                  "results": Array [],
                                                }
                              }
                              textHandler={[Function]}
                    />
          </form>
</div>,
        <div
          style={
                    Object {
                              "display": "inline-block",
                              "verticalAlign": "top",
                              "width": "50%",
                            }
          }
>
          <ResultsHolder
                    capitalize={[Function]}
                    state={
                              Object {
                                        "noResults": false,
                                        "params": Object {
                                          "fuel": "Any",
                                          "location": "London, UK",
                                          "max_distance": 50,
                                          "number_of_months": 12,
                                          "number_of_weeks": 52,
                                          "order_by": "price",
                                          "order_direction": "asc",
                                          "page": 1,
                                          "per_page": 15,
                                          "price_max": 2500,
                                          "price_min": 100,
                                          "rolling": false,
                                          "subscription_start_days": 30,
                                          "tags": "Any",
                                          "transmission": "Any",
                                          "vehicle_make": "Any",
                                          "vehicle_type": "Consumer",
                                          "year": 2018,
                                        },
                                        "results": Array [],
                                      }
                    }
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Logo
              title="Drover"
/>,
            <form>
              <InputsHolder
                            capitalize={[Function]}
                            scaleHandler={[Function]}
                            state={
                                          Object {
                                                        "noResults": false,
                                                        "params": Object {
                                                          "fuel": "Any",
                                                          "location": "London, UK",
                                                          "max_distance": 50,
                                                          "number_of_months": 12,
                                                          "number_of_weeks": 52,
                                                          "order_by": "price",
                                                          "order_direction": "asc",
                                                          "page": 1,
                                                          "per_page": 15,
                                                          "price_max": 2500,
                                                          "price_min": 100,
                                                          "rolling": false,
                                                          "subscription_start_days": 30,
                                                          "tags": "Any",
                                                          "transmission": "Any",
                                                          "vehicle_make": "Any",
                                                          "vehicle_type": "Consumer",
                                                          "year": 2018,
                                                        },
                                                        "results": Array [],
                                                      }
                            }
                            textHandler={[Function]}
              />
</form>,
          ],
          "style": Object {
            "display": "inline-block",
            "verticalAlign": "top",
            "width": "50%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "Drover",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <InputsHolder
                capitalize={[Function]}
                scaleHandler={[Function]}
                state={
                                Object {
                                                "noResults": false,
                                                "params": Object {
                                                  "fuel": "Any",
                                                  "location": "London, UK",
                                                  "max_distance": 50,
                                                  "number_of_months": 12,
                                                  "number_of_weeks": 52,
                                                  "order_by": "price",
                                                  "order_direction": "asc",
                                                  "page": 1,
                                                  "per_page": 15,
                                                  "price_max": 2500,
                                                  "price_min": 100,
                                                  "rolling": false,
                                                  "subscription_start_days": 30,
                                                  "tags": "Any",
                                                  "transmission": "Any",
                                                  "vehicle_make": "Any",
                                                  "vehicle_type": "Consumer",
                                                  "year": 2018,
                                                },
                                                "results": Array [],
                                              }
                }
                textHandler={[Function]}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "capitalize": [Function],
                "scaleHandler": [Function],
                "state": Object {
                  "noResults": false,
                  "params": Object {
                    "fuel": "Any",
                    "location": "London, UK",
                    "max_distance": 50,
                    "number_of_months": 12,
                    "number_of_weeks": 52,
                    "order_by": "price",
                    "order_direction": "asc",
                    "page": 1,
                    "per_page": 15,
                    "price_max": 2500,
                    "price_min": 100,
                    "rolling": false,
                    "subscription_start_days": 30,
                    "tags": "Any",
                    "transmission": "Any",
                    "vehicle_make": "Any",
                    "vehicle_type": "Consumer",
                    "year": 2018,
                  },
                  "results": Array [],
                },
                "textHandler": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "form",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ResultsHolder
            capitalize={[Function]}
            state={
                        Object {
                                    "noResults": false,
                                    "params": Object {
                                      "fuel": "Any",
                                      "location": "London, UK",
                                      "max_distance": 50,
                                      "number_of_months": 12,
                                      "number_of_weeks": 52,
                                      "order_by": "price",
                                      "order_direction": "asc",
                                      "page": 1,
                                      "per_page": 15,
                                      "price_max": 2500,
                                      "price_min": 100,
                                      "rolling": false,
                                      "subscription_start_days": 30,
                                      "tags": "Any",
                                      "transmission": "Any",
                                      "vehicle_make": "Any",
                                      "vehicle_type": "Consumer",
                                      "year": 2018,
                                    },
                                    "results": Array [],
                                  }
            }
/>,
          "style": Object {
            "display": "inline-block",
            "verticalAlign": "top",
            "width": "50%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "capitalize": [Function],
            "state": Object {
              "noResults": false,
              "params": Object {
                "fuel": "Any",
                "location": "London, UK",
                "max_distance": 50,
                "number_of_months": 12,
                "number_of_weeks": 52,
                "order_by": "price",
                "order_direction": "asc",
                "page": 1,
                "per_page": 15,
                "price_max": 2500,
                "price_min": 100,
                "rolling": false,
                "subscription_start_days": 30,
                "tags": "Any",
                "transmission": "Any",
                "vehicle_make": "Any",
                "vehicle_type": "Consumer",
                "year": 2018,
              },
              "results": Array [],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            style={
                        Object {
                                    "display": "inline-block",
                                    "verticalAlign": "top",
                                    "width": "50%",
                                  }
            }
>
            <Logo
                        title="Drover"
            />
            <form>
                        <InputsHolder
                                    capitalize={[Function]}
                                    scaleHandler={[Function]}
                                    state={
                                                Object {
                                                            "noResults": false,
                                                            "params": Object {
                                                              "fuel": "Any",
                                                              "location": "London, UK",
                                                              "max_distance": 50,
                                                              "number_of_months": 12,
                                                              "number_of_weeks": 52,
                                                              "order_by": "price",
                                                              "order_direction": "asc",
                                                              "page": 1,
                                                              "per_page": 15,
                                                              "price_max": 2500,
                                                              "price_min": 100,
                                                              "rolling": false,
                                                              "subscription_start_days": 30,
                                                              "tags": "Any",
                                                              "transmission": "Any",
                                                              "vehicle_make": "Any",
                                                              "vehicle_type": "Consumer",
                                                              "year": 2018,
                                                            },
                                                            "results": Array [],
                                                          }
                                    }
                                    textHandler={[Function]}
                        />
            </form>
</div>,
          <div
            style={
                        Object {
                                    "display": "inline-block",
                                    "verticalAlign": "top",
                                    "width": "50%",
                                  }
            }
>
            <ResultsHolder
                        capitalize={[Function]}
                        state={
                                    Object {
                                                "noResults": false,
                                                "params": Object {
                                                  "fuel": "Any",
                                                  "location": "London, UK",
                                                  "max_distance": 50,
                                                  "number_of_months": 12,
                                                  "number_of_weeks": 52,
                                                  "order_by": "price",
                                                  "order_direction": "asc",
                                                  "page": 1,
                                                  "per_page": 15,
                                                  "price_max": 2500,
                                                  "price_min": 100,
                                                  "rolling": false,
                                                  "subscription_start_days": 30,
                                                  "tags": "Any",
                                                  "transmission": "Any",
                                                  "vehicle_make": "Any",
                                                  "vehicle_type": "Consumer",
                                                  "year": 2018,
                                                },
                                                "results": Array [],
                                              }
                        }
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Logo
                title="Drover"
/>,
              <form>
                <InputsHolder
                                capitalize={[Function]}
                                scaleHandler={[Function]}
                                state={
                                                Object {
                                                                "noResults": false,
                                                                "params": Object {
                                                                  "fuel": "Any",
                                                                  "location": "London, UK",
                                                                  "max_distance": 50,
                                                                  "number_of_months": 12,
                                                                  "number_of_weeks": 52,
                                                                  "order_by": "price",
                                                                  "order_direction": "asc",
                                                                  "page": 1,
                                                                  "per_page": 15,
                                                                  "price_max": 2500,
                                                                  "price_min": 100,
                                                                  "rolling": false,
                                                                  "subscription_start_days": 30,
                                                                  "tags": "Any",
                                                                  "transmission": "Any",
                                                                  "vehicle_make": "Any",
                                                                  "vehicle_type": "Consumer",
                                                                  "year": 2018,
                                                                },
                                                                "results": Array [],
                                                              }
                                }
                                textHandler={[Function]}
                />
</form>,
            ],
            "style": Object {
              "display": "inline-block",
              "verticalAlign": "top",
              "width": "50%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "Drover",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <InputsHolder
                  capitalize={[Function]}
                  scaleHandler={[Function]}
                  state={
                                    Object {
                                                      "noResults": false,
                                                      "params": Object {
                                                        "fuel": "Any",
                                                        "location": "London, UK",
                                                        "max_distance": 50,
                                                        "number_of_months": 12,
                                                        "number_of_weeks": 52,
                                                        "order_by": "price",
                                                        "order_direction": "asc",
                                                        "page": 1,
                                                        "per_page": 15,
                                                        "price_max": 2500,
                                                        "price_min": 100,
                                                        "rolling": false,
                                                        "subscription_start_days": 30,
                                                        "tags": "Any",
                                                        "transmission": "Any",
                                                        "vehicle_make": "Any",
                                                        "vehicle_type": "Consumer",
                                                        "year": 2018,
                                                      },
                                                      "results": Array [],
                                                    }
                  }
                  textHandler={[Function]}
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "capitalize": [Function],
                  "scaleHandler": [Function],
                  "state": Object {
                    "noResults": false,
                    "params": Object {
                      "fuel": "Any",
                      "location": "London, UK",
                      "max_distance": 50,
                      "number_of_months": 12,
                      "number_of_weeks": 52,
                      "order_by": "price",
                      "order_direction": "asc",
                      "page": 1,
                      "per_page": 15,
                      "price_max": 2500,
                      "price_min": 100,
                      "rolling": false,
                      "subscription_start_days": 30,
                      "tags": "Any",
                      "transmission": "Any",
                      "vehicle_make": "Any",
                      "vehicle_type": "Consumer",
                      "year": 2018,
                    },
                    "results": Array [],
                  },
                  "textHandler": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "form",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ResultsHolder
              capitalize={[Function]}
              state={
                            Object {
                                          "noResults": false,
                                          "params": Object {
                                            "fuel": "Any",
                                            "location": "London, UK",
                                            "max_distance": 50,
                                            "number_of_months": 12,
                                            "number_of_weeks": 52,
                                            "order_by": "price",
                                            "order_direction": "asc",
                                            "page": 1,
                                            "per_page": 15,
                                            "price_max": 2500,
                                            "price_min": 100,
                                            "rolling": false,
                                            "subscription_start_days": 30,
                                            "tags": "Any",
                                            "transmission": "Any",
                                            "vehicle_make": "Any",
                                            "vehicle_type": "Consumer",
                                            "year": 2018,
                                          },
                                          "results": Array [],
                                        }
              }
/>,
            "style": Object {
              "display": "inline-block",
              "verticalAlign": "top",
              "width": "50%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "capitalize": [Function],
              "state": Object {
                "noResults": false,
                "params": Object {
                  "fuel": "Any",
                  "location": "London, UK",
                  "max_distance": 50,
                  "number_of_months": 12,
                  "number_of_weeks": 52,
                  "order_by": "price",
                  "order_direction": "asc",
                  "page": 1,
                  "per_page": 15,
                  "price_max": 2500,
                  "price_min": 100,
                  "rolling": false,
                  "subscription_start_days": 30,
                  "tags": "Any",
                  "transmission": "Any",
                  "vehicle_make": "Any",
                  "vehicle_type": "Consumer",
                  "year": 2018,
                },
                "results": Array [],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
